storage/src/test/java/tech/pegasys/teku/storage/client/RecentChainDataTest.java

```diff
@@ -251,6 +253,90 @@ public void updateHead_blockAndStateAreMissing() {
         .contains(genesis.getRoot());
   }
 
+  @Test
+  void createHead_upOnStatingWithNonTransitionedState()
+      throws SlotProcessingException, EpochProcessingException {
+    initPreGenesis();
+    final long lastNonEmptyBlockSlot = 7;
+    final ChainBuilder chainBuilder = ChainBuilder.create(spec);
+    final UInt64 genesisTime = UInt64.valueOf(1000);
+    chainBuilder.generateGenesis(genesisTime, true);
+    chainBuilder.generateBlocksUpToSlot(7);
+
+    final SignedBlockAndState lastState = chainBuilder.getBlockAndStateAtSlot(7);
+    final AnchorPoint anchorPoint = AnchorPoint.fromInitialState(spec, lastState.getState());
+    recentChainData.initializeFromAnchorPoint(anchorPoint, UInt64.valueOf(10000));
+
+    assertThat(recentChainData.getChainHead().map(MinimalBeaconBlockSummary::getRoot))
+        .contains(anchorPoint.getRoot());
+    assertThat(recentChainData.getChainHead().map(MinimalBeaconBlockSummary::getSlot))
+        .contains(UInt64.valueOf(lastNonEmptyBlockSlot));
+  }
+
+  @Test
+  void createHead_upOnStatingWithTransitionedStateWithEmptyBlock()
+      throws SlotProcessingException, EpochProcessingException {
+    initPreGenesis();
+    final ChainBuilder chainBuilder = ChainBuilder.create(spec);
+    final UInt64 genesisTime = UInt64.valueOf(1000);
+    chainBuilder.generateGenesis(genesisTime, true);
+    chainBuilder.generateBlocksUpToSlot(7);
+
+    final SignedBlockAndState lastState = chainBuilder.getBlockAndStateAtSlot(7);
+    final BeaconState transitionedState =
+        spec.processSlots(lastState.getState(), UInt64.valueOf(8));
+    final AnchorPoint anchorPoint = AnchorPoint.fromInitialState(spec, transitionedState);
+    recentChainData.initializeFromAnchorPoint(anchorPoint, UInt64.valueOf(10000));
+    // blockRoot should be from last non-empty block
+    assertThat(recentChainData.getChainHead().map(MinimalBeaconBlockSummary::getRoot))
+        .contains(anchorPoint.getRoot());
+    // head should be
```

Comment:
expand comments to be readable or remove...

Reply:
fixed comment